pkg/datapath/loader/netlink.go

```diff
@@ -329,3 +332,57 @@ func SetupBaseDevice(mtu int) (netlink.Link, netlink.Link, error) {
 
 	return linkHost, linkNet, nil
 }
+
+// reloadIPSecOnLinkChanges subscribes to link changes to detect newly added devices
+// and reinitializes IPsec on changes. Only in effect for ENI mode in which we expect
+// new devices at runtime.
+func (l *Loader) reloadIPSecOnLinkChanges() {
+	// settleDuration is the amount of time to wait for further link updates
+	// before proceeding with reinitialization. This avoids reinitializing
+	// twice when seeing NEWLINK followed by SETLINK.
+	const settleDuration = 5 * time.Second
```

Comment:
That seems like a fairly long time. Couldn't we instead filter out events we don't want to see (maybe `u.Header.Type == unix.RTM_NEWLINK`)?

Reply:
Since this is also triggering compilation of the bpf_network program I wanted to err on the side of caution.

Reply:
Changed this now so that the delay/coalescing is after the first reinitialization so there won't be any delay. I've also moved the call to recompile `bpf_network.c` into `Reinitialize` so it won't be performed every time.

Reply:
> I've also moved the call to recompile bpf_network.c into Reinitialize so it won't be performed every time.

I hadn't considered that. Nice!

> Changed this now so that the delay/coalescing is after the first reinitialization so there won't be any delay.

Did you check that loading the BPF program right after the NEWLINK and without waiting for SETLINK works fine? I'm guessing if we don't see any errors/warnings when testing this code, it means it's okay, no?